src/mongo/mongo.ts

```diff
@@ -40,18 +40,28 @@ export class Model implements IMongoResource {
 	readonly type: string = 'mongoRoot';
 	readonly canHaveChildren: boolean = true;
 
-	private _azureServers: IMongoResource[] = [];
-	private _servers: IMongoResource[] = [];
+	private _azureServers: Server[] = [];
+	private _servers: Server[] = [];
 	private _isLoading: boolean = false;
 	
 	private _onChange: EventEmitter<void> = new EventEmitter<void>();
 	readonly onChange: Event<void> = this._onChange.event;
 
+	private readonly _credentialsService = "ms-azuretools.vscode-cosmosdb";
+	private readonly _credentialsAccount = "mongoConnectionStrings";
+
 	constructor(private azureAccount: AzureAccount) {
+		this.azureAccount.credentials.readSecret(this._credentialsService, this._credentialsAccount).then(async value => {
+			const connectionStrings: string[] = value ? JSON.parse(value) : null;
```

Comment:
What happens if _value_ is not a valid JSON string? Although unlikely, the secret can be manually modified from the Control Panel on Windows (not sure about other platforms) or by any other apps using the same underlying Win32 functions.

Reply:
Wrapped in a try/catch with an error message